tools/old_config_maker/metric.py

```diff
@@ -0,0 +1,243 @@
+# Wraps around 2DLine/plt.text object
+
+from math import *
+
+
+class Metric:
+    """
+    Piece of data to be tracked.
+
+    Parameters
+    ----------
+    output: Line2D(plt animation) or text(plt/ax.text)
+        Where to output data
+    func: string
+        The function, as string, to generate values(takes variables x, y, z corresponding to data streams)
+    x_stream: string, optional unless x used in func
+        Data from drone to serve as x variable in function
+    y_stream: string, optional unless y used in func
+        Data from drone to serve as y variable in function
+    z_stream: string, optional unless z used in func
+        Data from drone to serve as z variable in function
+    """
+
+    possible_data_streams = [
+        'altitude', 'airspeed', 'velocity_x', 'velocity_y', 'velocity_z', 'voltage', 'state', 'mode', 'armed',
+        'roll', 'pitch', 'yaw', 'altitude_controller_output', 'altitude_rc_output', 'target_altitude',
+        'pitch_controller_output', 'pitch_rc_output', 'target_pitch_velocity', 'roll_controller_output',
+        'roll_rc_output', 'target_roll_velocity', 'yaw_controller_output', 'yaw_rc_output', 'target_yaw',
+        'color_image', 'depth_image', None]
+
+    def __init__(self, output, label=None, func=None, x_stream=None, y_stream=None, z_stream=None):
+        self._output = output
+
+        try:
+            self._label = output.label() if not label else label
+        except AttributeError:
+            self._label = label
+
+        #
+        # Set func
+        self._raw_func = func if func else 'x'
+
+        # Func safety check
+        for letter in self._raw_func:
+            assert letter in ',.0123456789 xyz()+-*/%absintfloatsincostanhlogp', \
```

Comment:
Use constants determined here https://docs.python.org/3.1/library/string.html
Also make lists of supported operations and join the strings together